### ISSUE TITLE
HTTP3 over local mode: Unable to access certain websites using firefox via HTTP3

### ISSUE THREAD

**Issue by errorxyz** [mitmproxy/mitmproxy#7117]
#### Problem Description

Firefox fails to access `cloudflare-quic.com` over HTTP3 while using mitmproxy in local mode. Other websites such as `http3.is` load up normally using HTTP3

#### Steps to reproduce the behavior:
1. Start local mode: `mitmdump --mode local:firefox`
2. Start firefox and access `http3.is` to ensure HTTP3 works
3. Accessing `cloudflare-quic.com` does not work over HTTP3

#### System Information
Mitmproxy: 11.0.0.dev (+18, commit 6bb536e)
Python:    3.11.3
OpenSSL:   OpenSSL 3.2.2 4 Jun 2024
Platform:  Windows-10-10.0.22631-SP0

**Comment by errorxyz** [comment]
Proxy debug: https://gist.github.com/errorxyz/f0c5cb1fa8c38daa5ac8548add665ff8

**Comment by errorxyz** [comment]
Proxy debug: https://gist.github.com/errorxyz/0f81b5d8ef3d1921e1b2deb08c24e990

**Comment by mhils** [comment]
I can't repro this here - cloudflare-quic.com works just fine both in local and redirect mode. It works just fine for you in reverse mode, only local mode is affected?

**Comment by errorxyz** [comment]
> I can't repro this here - cloudflare-quic.com works just fine both in local and redirect mode. It works just fine for you in reverse mode, only local mode is affected?

Yes, works fine in reverse mode but doesn't work if we're using local mode
